stdlib/public/Concurrency/Task.cpp

```diff
@@ -1656,7 +1657,7 @@ static void swift_task_asyncMainDrainQueueImpl() {
   }
 
   pfndispatch_main();
-  swift_unreachable("Returned from dispatch_main()");
+  swift_Concurrency_fatalError("Returned from dispatch_main()");
```

Comment:
They the change from `swift_unreachable`?  That seems like a much better solution as it tells the compiler that the path is dead.